packages/components/atoms/f-form-field/src/components/FormField.vue

```diff
@@ -260,6 +290,18 @@ export default {
 
         isDisabled () {
             return this.$attrs.disabled === 'disabled';
+        },
+
+        isValidIconField () {
+            return VALID_ICON_INPUT_TYPES.includes(this.inputType);
```

Comment:
Would be good to throw an error if someone is trying to use an icon on an invalid field type.

I've done something similar [when validating props](https://github.com/justeat/fozzie-components/blob/master/packages/components/atoms/f-button/src/components/Button.vue#L130-L133) – this is sightly different, but I wonder if you could set a watcher on this computed prop and if it's invalid, throw a type error.

Would mean you can not really worry about checking for `isValidIconField` then below as it'll have thrown an error and won't render at all until resolved.